packages/compass-e2e-tests/helpers/commands/workspace-tabs.ts

```diff
@@ -48,17 +49,21 @@ export async function closeWorkspaceTabs(
     return (await browser.$$(Selectors.workspaceTab())).length;
   };
 
-  while ((await countTabs()) > 0) {
-    const currentActiveTab = await browser.$(
-      Selectors.workspaceTab({ active: true })
-    );
-    await currentActiveTab.click();
-    // Close this exact active tab rather than "the active one" because if there
-    // are multiple tabs then another tab will immediately become active and
-    // trip up the logic that checks that the tab you closed went away.
-    const id = await currentActiveTab.getAttribute('id');
-    await closeTab(browser, { id }, autoConfirmTabClose);
-  }
+  await browser.waitUntil(async () => {
```

Comment:
waitUntil is the better way to poll/repeat than a tight loop. Takes into account proper timeouts.